File: scripts/sql_queries.py
"""
SQL query loader for IMDB import script.
Provides clean separation between SQL and Python code.
"""

import os
from pathlib import Path
from typing import Optional

SQL_DIR: Path = Path(__file__).parent.parent / "sql"

def load_sql(filename: str) -> str:
    """Load SQL from a file in the sql directory."""
    sql_path: Path = SQL_DIR / filename
    if not sql_path.exists():
        raise FileNotFoundError(f"SQL file not found: {sql_path}")
    
    return sql_path.read_text(encoding='utf-8').strip()

class SQLQueries:
    """Container for all SQL queries used in the import process."""
    
    def __init__(self, sql_dir: Optional[str] = None) -> None:
        if sql_dir is None:
            self.sql_dir: Path = Path(__file__).parent.parent / "sql"
        else:
            self.sql_dir = Path(sql_dir)
    
    def load(self, filename: str) -> str:
        """Load SQL from a file."""
        sql_path: Path = self.sql_dir / filename
        return sql_path.read_text(encoding='utf-8').strip()
    
    @property
    def temp_tables(self) -> str:
        return self.load("temp_tables.sql")
    
    @property
    def import_titles(self) -> str:
        return self.load("import_titles.sql")
    
    @property
    def import_persons(self) -> str:
        return self.load("import_persons.sql")
    
    @property
    def import_ratings(self) -> str:
        return self.load("import_ratings.sql")
    
    @property
    def import_episodes(self) -> str:
        return self.load("import_episodes.sql")
    
    @property
    def import_principals(self) -> str:
        return self.load("import_principals.sql")
    
    @property
    def import_crew_directors(self) -> str:
        return self.load("import_crew.sql")
    
    @property
    def import_crew_writers(self) -> str:
        return self.load("import_crew_writers.sql")

# Global instance for easy access
queries: SQLQueries = SQLQueries() 

File: scripts/sql-queries.ts
import path from "node:path";
import { fileURLToPath } from "node:url";
import fs from "fs-extra";

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);

const SQL_DIR = path.join(__dirname, "..", "sql");

export async function loadSql(filename: string): Promise<string> {
  const sqlPath = path.join(SQL_DIR, filename);
  if (!(await fs.pathExists(sqlPath))) {
    throw new Error(`SQL file not found: ${sqlPath}`);
  }

  return (await fs.readFile(sqlPath, "utf-8")).trim();
}

export class SQLQueries {
  private sqlDir: string;

  constructor(sqlDir?: string) {
    this.sqlDir = sqlDir || SQL_DIR;
  }

  async load(filename: string): Promise<string> {
    const sqlPath = path.join(this.sqlDir, filename);
    return (await fs.readFile(sqlPath, "utf-8")).trim();
  }

  async getTempTables(): Promise<string> {
    return this.load("temp_tables.sql");
  }

  async getImportTitles(): Promise<string> {
    return this.load("import_titles.sql");
  }

  async getImportPersons(): Promise<string> {
    return this.load("import_persons.sql");
  }

  async getImportRatings(): Promise<string> {
    return this.load("import_ratings.sql");
  }

  async getImportEpisodes(): Promise<string> {
    return this.load("import_episodes.sql");
  }

  async getImportPrincipals(): Promise<string> {
    return this.load("import_principals.sql");
  }

  async getImportCrewDirectors(): Promise<string> {
    return this.load("import_crew.sql");
  }

  async getImportCrewWriters(): Promise<string> {
    return this.load("import_crew_writers.sql");
  }
}

// Global instance for easy access
export const queries = new SQLQueries();
